compiler_tester/src/directories/ethereum/test.rs

```diff
@@ -296,7 +296,7 @@ impl Buildable for EthereumTest {
         let last_source = self.last_source(summary.clone(), &mode)?;
 
         let (contract_address, libraries_addresses, libraries) = match self.get_addresses(
-            EVMAddressIterator::new(matches!(target, Target::EVM)),
+            EVMAddressIterator::new(false),
```

Comment:
Could you remove this argument?
It was added to overcome the old emulator bug, which is not used anymore.

Reply:
After looking into this for a while, I realized that moving this to be always `false` broke the interpreter benchmarks.
To fix it, I had to change it so the iterator receives by argument the initial value for the nonce here [3a374a3](https://github.com/matter-labs/era-compiler-tester/pull/75/commits/3a374a33407669c45ec300cd7829f69345334ee0). I'll explain the full reasons below:

The change we had to do to the `ContractDeployer` involved modifying the `createEVM` function so that the deployment nonce essentially started at one instead of zero when using it for the purposes of deploying (i.e. to calculate the deployed contract's address as`hash(deployer_address, nonce)`). This is not a great solution actually, because it does not really change nonces, it just pretends that the deployer's nonce is one greater than it actually is.

The reason this is not a big deal is that the `createEVM` function is really only used by the `compiler-tester` to deploy contracts on tests, it's never used by the EVM interpreter in any way (it uses `createEVMInternal` instead, which receives the address of the contract to deploy instead). So really the change is just saying "the compiler tester should start nonces at 1 when deploying", which is in line with what the solidity tests expect.

To accomodate to this change, the `EVMAddressIterator` had to be modified as well to start at `1`. This works, but breaks the interpreter benchmarks. For some reason, these benchmarks start the nonce at a value one greater than the solidity ones, so after changing the `ContractDeployer` to start at `1` we need to start it at `2` (beforehand they worked because the target was `EraVM` for these tests, not EraVM, so the address iterator started at 1 instead of zero as was the case for `EVM`).

There might be a better solution to all this where we don't modify the `ContractDeployer` and instead do all nonce calculations on the `compiler-tester` and call `createEVMInternal` for deployment, but for now I believe this works.